src/main/scala/org/clulab/habitus/actions/HabitusActions.scala

```diff
@@ -170,5 +170,45 @@ class HabitusActions extends Actions {
       )
     }
   }
+  def splitIntoBinary(mentions: Seq[Mention]): Seq[Mention] = {
+    val toReturn = new ArrayBuffer[Mention]()
+    val (target, other) = mentions.partition( m => {
+      val valueLabels = m.arguments("value").map(_.label)
+      valueLabels.length > 1 &&
+      valueLabels.distinct.length == 1
+    }
+    )
+    for (m <- target) {
+      for (v <- m.arguments("value")) {
+        val newArgs = Map("variable" -> m.arguments("variable"), "value" -> Seq(v))
+        val newMention = copyWithArgs(m, newArgs = newArgs)
+        toReturn.append(newMention)
+      }
+    }
+    toReturn ++ other
```

Comment:
I see an ArrayBuffer, which always puts me on edge.  You might consider something like this which Scala does surprisingly nicely.

```scala
    val splitTargets = for {
      m <- targets
      value <- m.arguments("value")
      newArgs = Map("variable" -> m.arguments("variable"), "value" -> Seq(value))
    } yield copyWithArgs(m, newArgs)
    splitTargets ++ other
```

Reply:
Yeah, I remembered to use for/yield in one of today's methods based on your previous comments regarding array buffers, but did not think of it here. To be honest, I hate these for/yield things because I can't print out inside of them :D But I am sure I will phase out my love for array buffers soon :D

Reply:
That's a very good point about the printing, so good that I looked it up once.  Will this work for you?

```scala
    val splitTargets = for {
      m <- targets
      _ = println(m)
      value <- m.arguments("value")
      _ = println(value)
      newArgs = Map("variable" -> m.arguments("variable"), "value" -> Seq(value))
      _ = println(newArgs)
    } yield copyWithArgs(m, newArgs)
```

Reply:
Neat! Did not know! Thanks!